src/Item/MenuItem.php

```diff
@@ -735,49 +772,13 @@ public function resolveTree (string $currentClass = "current", string $ancestorC
             $this->addListItemClass($ancestorClass);
         }
 
-        // sort by priority
-        $this->children = $this->sort($this->children);
+        // sort children
+        $this->children = MenuItemSorter::sort($this->sort, $this->children);
```

Comment:
I'm not too sure about this static dependency here as I generally dislike static method calls inside a plain PHP object like this one. Can't we pass in an item sorter whenever we want to resolve the tree? The caller could then decide which sorting method should be used by simply passing in a different one. That way applications could easily change the sorting implementation, if necessary - although, to be fair, I think we already cover a lot of use cases with the current sorter.

So the `$option["sort"]` could be as it's replaced by a dedicated sorter. This method signature would then look like `public function resolveTree (string $currentClass = "current", string $ancestorClass = "ancestor", int $level = 0, `?MenuItemSorterInterface $itemSorter = null) : bool`

Reply:
The point here is that the sorter should be a simple function without any dependencies.  We could in-line it, but that just bloats the menu item. We could make it non-static, but that doesn’t give us anything except worse performance and GC pressure. We could make it a real function, but then autoloading is sort-of broken. 

Passing it in doesn’t work, as the resolveTree call is recursive and every item can have its own sorting (that is the point of this PR). 

So passing a sorter in doesn’t work, we could pass a registry of sorters in. 

But...

a) that is brutally over-engineered 😕
b) that part is quite performance-sensitive and that would just kill all of it. Keep in mind that static calls are super fast in PHP.